src/HotChocolate/Core/src/Types.Scalars/IPv4Type.cs

```diff
@@ -1,114 +1,26 @@
 using System.Text.RegularExpressions;
-using HotChocolate.Language;
 
 namespace HotChocolate.Types.Scalars
 {
     /// <summary>
-    /// The `IPv4` scalar type represents a valid a IPv4 address as defined
-    /// here https://en.wikipedia.org/wiki/IPv4.
+    /// The `IPv4` scalar type represents a valid a IPv4 address as defined here
+    /// https://en.wikipedia.org/wiki/IPv4.
     /// </summary>
-    public class IPv4Type : StringType
+    public class IPv4Type : RegularExpressionType
     {
-        private static readonly string _validationPattern =
-            ScalarResources.IPv4Type_ValidationPattern;
-
-        private static readonly Regex _validationRegex =
-            new(_validationPattern, RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private const string _validationPattern =
```

Comment:
@michaelstaib  can you live with this?

Reply:
We can refactor it when we refactor all the RegEx Scalar Types..